Test failures on 32-bit architectures only

Running the tests of the last release, 2.4, fails on 32-bit architectures on Alpine Linux (x86 and armv7). It (luckily) works fine on any 64-bit architectures.

```
 _________________ TestConvertPandas.test_from_adjacency_named __________________
self = <networkx.tests.test_convert_pandas.TestConvertPandas object at 0xe5a23658>
    def test_from_adjacency_named(self):
        # example from issue #3105
        data = {"A": {"A": 0, "B": 0, "C": 0},
                "B": {"A": 1, "B": 0, "C": 0},
                "C": {"A": 0, "B": 1, "C": 0}}
        dftrue = pd.DataFrame(data)
        df = dftrue[["A", "C", "B"]]
        G = nx.from_pandas_adjacency(df, create_using=nx.DiGraph())
        df = nx.to_pandas_adjacency(G, dtype=int)
>       pd.testing.assert_frame_equal(df, dftrue)
E       AssertionError: Attributes of DataFrame.iloc[:, 0] (column name="A") are different
E       
E       Attribute "dtype" are different
E       [left]:  int32
E       [right]: int64
networkx/tests/test_convert_pandas.py:173: AssertionError
```

No that is just a warning. The remaining issue is the following:

```
 ____________________ TestWeightedPath.test_weight_functions ____________________
self = <networkx.algorithms.shortest_paths.tests.test_weighted.TestWeightedPath object at 0xe5d960a0>
    def test_weight_functions(self):
        def heuristic(*z):
            return hash(z)
    
        def getpath(pred, v, s):
            return [v] if v == s else getpath(pred, pred[v], s) + [v]
    
        def goldberg_radzik(g, s, t, weight="weight"):
            pred, dist = nx.goldberg_radzik(g, s, weight=weight)
            dist = dist[t]
            return dist, getpath(pred, t, s)
    
        def astar(g, s, t, weight="weight"):
            path = nx.astar_path(g, s, t, heuristic, weight=weight)
            dist = nx.astar_path_length(g, s, t, heuristic, weight=weight)
            return dist, path
    
        def vlp(G, s, t, l, F, w):
            res = F(G, s, t, weight=w)
            validate_length_path(G, s, t, l, *res, weight=w)
    
        G = self.cycle
        s = 6
        t = 4
        path = [6] + list(range(t + 1))
    
        def weight(u, v, _):
            return 1 + v ** 2
    
        length = sum(weight(u, v, None) for u, v in pairwise(path))
        vlp(G, s, t, length, nx.bidirectional_dijkstra, weight)
        vlp(G, s, t, length, nx.single_source_dijkstra, weight)
        vlp(G, s, t, length, nx.single_source_bellman_ford, weight)
        vlp(G, s, t, length, goldberg_radzik, weight)
>       vlp(G, s, t, length, astar, weight)
networkx/algorithms/shortest_paths/tests/test_weighted.py:233: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
networkx/algorithms/shortest_paths/tests/test_weighted.py:218: in vlp
    validate_length_path(G, s, t, l, *res, weight=w)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
G = <networkx.classes.graph.Graph object at 0xe5d969a0>, s = 6, t = 4
soln_len = 35, length = 43, path = [6, 5, 4]
weight = <function TestWeightedPath.test_weight_functions.<locals>.weight at 0xe612eda8>
    def validate_length_path(G, s, t, soln_len, length, path, weight="weight"):
>       assert soln_len == length
E       assert 35 == 43
networkx/algorithms/shortest_paths/tests/test_weighted.py:29: AssertionError
```

I'd say unrelated to the 32-bit problem, but also not because of pygraphviz.

--- a/networkx/tests/test_convert_pandas.py
+++ b/networkx/tests/test_convert_pandas.py
@@ -229,7 +229,7 @@ class TestConvertPandas:
             "B": {"A": 1, "B": 0, "C": 0},
             "C": {"A": 0, "B": 1, "C": 0},
         }
-        dftrue = pd.DataFrame(data)
+        dftrue = pd.DataFrame(data, dtype=np.intp)
         df = dftrue[["A", "C", "B"]]
         G = nx.from_pandas_adjacency(df, create_using=nx.DiGraph())
         df = nx.to_pandas_adjacency(G, dtype=np.intp)

